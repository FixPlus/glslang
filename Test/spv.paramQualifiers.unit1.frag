#version 460 core

float foo(in float f, out int i, inout vec3 v)
{
    v.x = f;
    i = int(v.y);
    return v.z;
}
