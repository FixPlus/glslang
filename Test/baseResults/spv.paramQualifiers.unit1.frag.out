spv.paramQualifiers.unit1.frag
Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:3  Function Definition: foo(f1;i1;vf3; ( global highp float)
0:3    Function Parameters: 
0:3      'f' ( in highp float)
0:3      'i' ( out highp int)
0:3      'v' ( inout highp 3-component vector of float)
0:5    Sequence
0:5      move second child to first child ( temp highp float)
0:5        direct index ( temp highp float)
0:5          'v' ( inout highp 3-component vector of float)
0:5          Constant:
0:5            0 (const int)
0:5        'f' ( in highp float)
0:6      move second child to first child ( temp highp int)
0:6        'i' ( out highp int)
0:6        Convert float to int ( temp highp int)
0:6          direct index ( temp highp float)
0:6            'v' ( inout highp 3-component vector of float)
0:6            Constant:
0:6              1 (const int)
0:7      Branch: Return with expression
0:7        direct index ( temp highp float)
0:7          'v' ( inout highp 3-component vector of float)
0:7          Constant:
0:7            2 (const int)
0:?   Linker Objects
0:?     Linker Function: foo(f1;i1;vf3; ( global highp float)
0:?       Function Parameters: 
0:?         '' ( in highp float)
0:?         '' ( out highp int)
0:?         '' ( inout highp 3-component vector of float)

spv.paramQualifiers.unit2.frag
Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:7  Function Definition: main( ( global void)
0:7    Function Parameters: 
0:?     Sequence
0:10      Sequence
0:10        move second child to first child ( temp highp 3-component vector of float)
0:10          'v' ( temp highp 3-component vector of float)
0:10          Constant:
0:10            0.000000
0:10            1.000000
0:10            2.000000
0:11      Sequence
0:11        move second child to first child ( temp highp float)
0:11          'r' ( temp highp float)
0:11          Function Call: foo(f1;i1;vf3; ( global highp float)
0:11            Constant:
0:11              1.000000
0:11            'x' ( temp highp int)
0:11            'v' ( temp highp 3-component vector of float)
0:?   Linker Objects
0:?     Linker Function: foo(f1;i1;vf3; ( global highp float)
0:?       Function Parameters: 
0:?         '' ( in highp float)
0:?         '' ( out highp int)
0:?         '' ( inout highp 3-component vector of float)
0:?     'cout' ( out highp float)


Linked fragment stage:


Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:3  Function Definition: foo(f1;i1;vf3; ( global highp float)
0:3    Function Parameters: 
0:3      'f' ( in highp float)
0:3      'i' ( out highp int)
0:3      'v' ( inout highp 3-component vector of float)
0:5    Sequence
0:5      move second child to first child ( temp highp float)
0:5        direct index ( temp highp float)
0:5          'v' ( inout highp 3-component vector of float)
0:5          Constant:
0:5            0 (const int)
0:5        'f' ( in highp float)
0:6      move second child to first child ( temp highp int)
0:6        'i' ( out highp int)
0:6        Convert float to int ( temp highp int)
0:6          direct index ( temp highp float)
0:6            'v' ( inout highp 3-component vector of float)
0:6            Constant:
0:6              1 (const int)
0:7      Branch: Return with expression
0:7        direct index ( temp highp float)
0:7          'v' ( inout highp 3-component vector of float)
0:7          Constant:
0:7            2 (const int)
0:7  Function Definition: main( ( global void)
0:7    Function Parameters: 
0:?     Sequence
0:10      Sequence
0:10        move second child to first child ( temp highp 3-component vector of float)
0:10          'v' ( temp highp 3-component vector of float)
0:10          Constant:
0:10            0.000000
0:10            1.000000
0:10            2.000000
0:11      Sequence
0:11        move second child to first child ( temp highp float)
0:11          'r' ( temp highp float)
0:11          Function Call: foo(f1;i1;vf3; ( global highp float)
0:11            Constant:
0:11              1.000000
0:11            'x' ( temp highp int)
0:11            'v' ( temp highp 3-component vector of float)
0:?   Linker Objects
0:?     'cout' ( out highp float)

// Module Version 10000
// Generated by (magic number): 80008
// Id's are bound by 47

                              Capability Shader
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Fragment 4  "main" 46
                              ExecutionMode 4 OriginUpperLeft
                              Source GLSL 460
                              Name 4  "main"
                              Name 16  "foo(f1;i1;vf3;"
                              Name 13  "f"
                              Name 14  "i"
                              Name 15  "v"
                              Name 31  "v"
                              Name 36  "r"
                              Name 37  "x"
                              Name 38  "param"
                              Name 39  "param"
                              Name 40  "param"
                              Name 46  "cout"
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeFloat 32
               7:             TypePointer Function 6(float)
               8:             TypeInt 32 1
               9:             TypePointer Function 8(int)
              10:             TypeVector 6(float) 3
              11:             TypePointer Function 10(fvec3)
              12:             TypeFunction 6(float) 7(ptr) 9(ptr) 11(ptr)
              19:             TypeInt 32 0
              20:     19(int) Constant 0
              22:     19(int) Constant 1
              26:     19(int) Constant 2
              32:    6(float) Constant 0
              33:    6(float) Constant 1065353216
              34:    6(float) Constant 1073741824
              35:   10(fvec3) ConstantComposite 32 33 34
              45:             TypePointer Output 6(float)
        46(cout):     45(ptr) Variable Output
         4(main):           2 Function None 3
               5:             Label
           31(v):     11(ptr) Variable Function
           36(r):      7(ptr) Variable Function
           37(x):      9(ptr) Variable Function
       38(param):      7(ptr) Variable Function
       39(param):      9(ptr) Variable Function
       40(param):     11(ptr) Variable Function
                              Store 31(v) 35
                              Store 38(param) 33
              41:   10(fvec3) Load 31(v)
                              Store 40(param) 41
              42:    6(float) FunctionCall 16(foo(f1;i1;vf3;) 38(param) 39(param) 40(param)
              43:      8(int) Load 39(param)
                              Store 37(x) 43
              44:   10(fvec3) Load 40(param)
                              Store 31(v) 44
                              Store 36(r) 42
                              Return
                              FunctionEnd
16(foo(f1;i1;vf3;):    6(float) Function None 12
           13(f):      7(ptr) FunctionParameter
           14(i):      9(ptr) FunctionParameter
           15(v):     11(ptr) FunctionParameter
              17:             Label
              18:    6(float) Load 13(f)
              21:      7(ptr) AccessChain 15(v) 20
                              Store 21 18
              23:      7(ptr) AccessChain 15(v) 22
              24:    6(float) Load 23
              25:      8(int) ConvertFToS 24
                              Store 14(i) 25
              27:      7(ptr) AccessChain 15(v) 26
              28:    6(float) Load 27
                              ReturnValue 28
                              FunctionEnd
